### ISSUE TITLE
Autocomplete self/cls in class methods

### ISSUE THREAD

**Issue by matthewbastien** [microsoft/pyright#3861]
**Is your feature request related to a problem? Please describe.**
Not necessarily a problem, but there are IDEs such as PyCharm that offer a very helpful feature that auto-completes class arguments for you so that you don't have to type the full signature. This reduces the amount of code one has to write manually and would be very helpful if brought to Pyright.

**Describe the solution you'd like**
When typing the opening parenthesis in the following class method, the `cls` argument is required and should be entered automatically by Pyright:
```python
class Foo:
    @classmethod
    def class_method(cls):
```

In this more basic method, the `self` argument is required and should be entered automatically by Pyright when the opening parenthesis is typed:
```python
class Bar:
    def instance_method(self):
```

In fact, it would be even be better to have Pyright complete the full class definition for you when you ask it for completions on `@classmethod`. Perhaps even when you type `def` there could be snippets (or some equivalent) that will write all of the basic method boilerplate for you and allow you to enter a method name.

**Comment by erictraut** [comment]
Since this isn't a core type checking feature request, I'm going to transfer it to the pylance issue tracker.
